django/utils/text.py

```diff
@@ -412,7 +413,12 @@ def slugify(value, allow_unicode=False):
     else:
         value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
     value = re.sub(r'[^\w\s-]', '', value).strip().lower()
-    return mark_safe(re.sub(r'[-\s]+', '-', value))
+
+    text = re.sub(r'[-\s\_]+', '-', value)
+    # remove redundant
+    text = re_duplicate_dash.sub("-", text).strip("-")
```

Comment:
Use single quotes (unless a string contains a single quote) as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).